clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -115,25 +114,30 @@ public int memberEpoch() {
 
     @Override
     public void updateState(ConsumerGroupHeartbeatResponseData response) {
-        if (response.errorCode() == Errors.NONE.code()) {
```

Comment:
I think we should validate here that the response contains no error (and throw IllegalArgument if so), as this func now is only expected to be called on successful responses. Without such validation, an erroneous call to this func in the case of an error would end up going unnoticed and transition the member to stable.